File: backend/app/schemas/issue.py
from typing import Any

from pydantic import BaseModel, ConfigDict

from app.schemas.review import ReviewDecisionSchema


class IssueSchema(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    analysis_run_id: int
    type: str
    severity: str
    description: str


class IssueEvidenceSchema(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    issue_id: int
    field_id: int
    page: int
    bbox: Any | None


class IssueEvidenceReadSchema(IssueEvidenceSchema):
    text: str


class IssueWithEvidencesSchema(IssueSchema):
    evidences: list[IssueEvidenceReadSchema]
    review: ReviewDecisionSchema | None = None
    review_status: str
    review_status_label: str

File: frontend/src/lib/types/issue.ts
import type {
  IssueReviewStatus,
  ReviewDecisionCode,
} from "@/lib/analysis/review-decisions";

export type IssueReview = {
  comment: string;
  decision: ReviewDecisionCode | string;
  decision_label: string;
  issue_id: number;
  status: IssueReviewStatus | string;
  status_label: string;
};

export type IssueEvidence = {
  bbox: Record<string, number> | null;
  field_id: number;
  issue_id: number;
  page: number;
  text: string;
};

export type AnalysisIssue = {
  analysis_run_id: number;
  description: string;
  evidences: IssueEvidence[];
  id: number;
  review: IssueReview | null;
  review_status: IssueReviewStatus | string;
  review_status_label: string;
  severity: string;
  type: string;
};

export type IssueBatchReviewResult = {
  comment: string;
  decision: string;
  decision_label: string;
  issue_ids: number[];
  updated_count: number;
};
